api/v1alpha1/postgresqlhostcredentials_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// PostgreSQLHostCredentialsSpec defines the desired state of PostgreSQLHostCredentials
+type PostgreSQLHostCredentialsSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Host is the hostname of the PostgreSQL instance.
+	Host string `json:"host,omitempty"`
+
+	// User is the admin user for the PostgreSQL instance. It will be used by
+	// posgresql-controller to manage resources on the host.
+	User ResourceVar `json:"user,omitempty"`
+
+	// Password is the admin user password for the PostgreSQL instance. It will
+	// be used by postgresql-controller to manage resources on the host.
+	Password ResourceVar `json:"password,omitempty"`
+
+	// Params is the space-separated list of parameters (e.g.,
```

Comment:
Is this really space separated?

Reply:
Probably not. I wrote this on the assumption that this project used the `host=... user=... sslmode=...` style of connection strings. I guess my preference would be to convert this into a `map[string]string` that the controller will convert into a url-encoded query string when it renders it to a connection string? This seems like it would be more convenient and less error prone than asking the user to pre-URL-encode? What are your thoughts?

Reply:
Right now this is just added to the connection string in an append way. I'm unsure what I'd prefer TBH. It seems wrong that this an untyped thing, I agree.

Our own usage is limited to just setting `sslmode=required` ATM so it would be easy enough to change to a map for us. The worst thing about it is really that PostgreSQL allows so many different ways of creating the connection string: https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-PARAMKEYWORDS

If we change this to a `map[string]string` type and then URI encode all values I guess we would cover most cases and avoid any pitfalls. But let's keep that in a separate PR to make progress on this.